src/containers/blocks.jsx

```diff
@@ -87,6 +87,7 @@ class Blocks extends React.Component {
     componentDidMount () {
         this.ScratchBlocks.FieldColourSlider.activateEyedropper_ = this.props.onActivateColorPicker;
         this.ScratchBlocks.Procedures.externalProcedureDefCallback = this.props.onActivateCustomProcedures;
+        this.ScratchBlocks.ScratchMsgs.setLocale(this.props.locale);
```

Comment:
It would be nice for `setLocale` to handle the special cases, but it doesn't appear to be simply checking `isVisible` in `setLocale` because it would still need to be called twice in `componentDidMount`, before the blocks are injected, and after the vm is attached.

Reply:
I'm not sure that it does need to be called after `attachVM`, it does appear to work, I'd want a lot more tests, like loading a project with an extension before making the change.